digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/ObstacleService.scala

```diff
@@ -34,14 +38,99 @@ class ObstacleService(val roadLinkService: RoadLinkService) extends PointAssetOp
     id
   }
 
+  override def getByBoundingBox(user: User, bounds: BoundingRectangle): Seq[PersistedAsset] = {
```

Comment:
It seems like there is a lot of duplicated code we can have a method on the base class that has a function as a parameter, and then reuse it.